cli/command/stack/kubernetes/ps.go

```diff
@@ -31,76 +37,73 @@ func RunPS(dockerCli *KubeCli, options options.PS) error {
 	}
 	podsClient := client.Pods()
 
-	// Fetch pods
-	if _, err := stacks.Get(namespace); err != nil {
-		return fmt.Errorf("nothing found in stack: %s", namespace)
+	filters := options.Filter.Value()
+	if err = filters.Validate(supportedPSFilters); err != nil {
+		return err
 	}
 
-	pods, err := fetchPods(namespace, podsClient)
+	// Fetch pods
+	if _, err := stacks.Get(stackName); err != nil {
+		return fmt.Errorf("nothing found in stack: %s", stackName)
+	}
+	pods, err := fetchPods(stackName, podsClient, filters)
 	if err != nil {
 		return err
 	}
 
 	if len(pods) == 0 {
-		return fmt.Errorf("nothing found in stack: %s", namespace)
+		return fmt.Errorf("nothing found in stack: %s", stackName)
 	}
+	return print(dockerCli, options, stackName, client, pods)
+}
 
+func print(dockerCli command.Cli, options options.PS, namespace string, client corev1.NodesGetter, pods []apiv1.Pod) error {
 	format := options.Format
-	if len(format) == 0 {
+	if format == "" {
 		format = task.DefaultFormat(dockerCli.ConfigFile(), options.Quiet)
 	}
-	nodeResolver := makeNodeResolver(options.NoResolve, client.Nodes())
 
 	tasks := make([]swarm.Task, len(pods))
 	for i, pod := range pods {
 		tasks[i] = podToTask(pod)
 	}
-	return print(dockerCli, namespace, tasks, pods, nodeResolver, !options.NoTrunc, options.Quiet, format)
-}
-
-type idResolver func(name string) (string, error)
-
-func print(dockerCli command.Cli, namespace string, tasks []swarm.Task, pods []apiv1.Pod, nodeResolver idResolver, trunc, quiet bool, format string) error {
 	sort.Stable(tasksBySlot(tasks))
 
 	names := map[string]string{}
 	nodes := map[string]string{}
 
-	tasksCtx := formatter.Context{
-		Output: dockerCli.Out(),
-		Format: formatter.NewTaskFormat(format, quiet),
-		Trunc:  trunc,
-	}
-
 	for i, task := range tasks {
-		nodeValue, err := nodeResolver(pods[i].Spec.NodeName)
+		nodeValue, err := resolveNode(options.NoResolve, client.Nodes(), pods[i].Spec.NodeName)
```

Comment:
should `client.Nodes()` be moved outside of the loop?

also, order of arguments is a bit odd; perhaps put `name` first